### PR TITLE
Release lifecycle docs overhaul & simplification

### PR DESCRIPTION
**Pull request by cwarnermm**
This docs PR introduces a simplification of our release documentation with a clear landing point: [Mattermost release life cycle](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/7110/guides/release-lifecycle.html)

From this page, you can reach the Mattermost [release policy](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/7110/lifecycle/release-policy.html), as well as [server](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/7110/lifecycle/mattermost-server-releases.html), [desktop](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/7110/lifecycle/mattermost-desktop-releases.html), and [mobile](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/7110/lifecycle/mattermost-mobile-releases.html) release details, including changelogs.

### CODE REVIEW

File: source/deploy/mattermost-v9-changelog.md [hunk 1]
```diff
@@ -0,0 +1,991 @@
+# Mattermost v9 changelog
+
+```{Important}
+**Support for Mattermost Server v8.1 Extended Support Release is coming to the end of its life cycle on May 15, 2024. Upgrading to Mattermost Server v9.5 or later is recommended.**
+- See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) documentation for details on upgrading to a newer release.
+- See the [changelog in progress](https://bit.ly/2nK3cVf) for details about the upcoming release.
+```
+
+(release-v9-8-feature-release)=
+## Release v9.8 - Feature Release
```
Comment (amyblais) [inline source/deploy/mattermost-v9-changelog.md:10]: Check doc links to the new Release types/definitions section.

File: source/lifecycle/version-archive.rst [hunk 2]
```diff
@@ -0,0 +1,742 @@
+Version Archive
+================
+
+.. include:: ../_static/badges/allplans-selfhosted.rst
+  :start-after: :nosearch:
+
+.. important::
+  - **Support for Mattermost Server v8.1 :doc:`Extended Support Release </upgrade/extended-support-release>` has come to the end of its life cycle on May 15, 2024. Upgrading to Mattermost Server v9.5 :doc:`Extended Support Release </upgrade/extended-support-release>` or later is required.**
```
Comment (amyblais) [inline source/lifecycle/version-archive.rst:8]: Check ESR and other links to new docs pages here and elsewhere.

**Review by amyblais — COMMENTED** [review]
Adding notes for things for me to remember to check when the draft is finalized.

File: source/lifecycle/client-compatibility.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Client compatibility
```
Comment (cwarnermm) [inline source/lifecycle/client-compatibility.md:1]: @nab-77 & @amyblais - Can I get your help to draft the client compatibility content we want published on this page, please?

File: source/lifecycle/mattermost-v9-changelog.md [hunk 4]
```diff
@@ -0,0 +1,991 @@
+# Mattermost v9 changelog
+
+```{Important}
+**Support for Mattermost Server v8.1 Extended Support Release has come to the end of its life cycle on May 15, 2024. Upgrading to Mattermost Server v9.5 or later is required.**
+- See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) documentation for details on upgrading to a newer release.
+- See the [changelog in progress](https://bit.ly/2nK3cVf) for details about the upcoming release.
+```
+
+(release-v9-8-feature-release)=
```
Comment (cwarnermm) [inline source/lifecycle/mattermost-v9-changelog.md:9]: @amyblais - Can I get your help to add the v9.9 draft changelog content here, please?

File: source/lifecycle/mattermost-mobile-releases.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+## Frequency
+
+Mattermost releases a new mobile app version every 4 months, in February, May, August, and November in binary form. Mattermost tests the latest mobile app release against the [latest server version](/lifecycle/mattermost-server-releases).
+
+See the [Mobile app changelog](/lifecycle/mobile-app-changelog) for release details, and see the [iOS mobile app](/collaborate/install-ios-app) and the [Android mobile app](/collaborate/install-android-app) documentation for installation details. 
+
+## Latest releases
+
+| **Release** | **Life cycle start** | **Life cycle end** |
```
Comment (cwarnermm) [inline source/lifecycle/mattermost-mobile-releases.md:18]: @amyblais - Can I get your help to populate the correct life cycle start and end dates for these mobile releases (based on our mobile changelog content) please?

File: source/lifecycle/mattermost-desktop-releases.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+
+```{Important}
+- From Mattermost v10, Mattermost server extended releases are now paired with Mattermost desktop app extended releases. See the [Mattermost extended support releases](#extended-support-releases) documentation to learn more about extended releases.
+- Automatic desktop app updates are supported and enabled on Windows, MacOS (via the Apple App Store), Ubuntu, Debian, and generic Linux platforms. See the [install Mattermost desktop app](/collaborate/install-desktop-app.html) documentation for platform-specific details on automatic app updates.
+```
+
+## Latest releases
+
+| **Release** | **Life cycle start** | **Life cycle end** |
```
Comment (cwarnermm) [inline source/lifecycle/mattermost-desktop-releases.md:22]: @amyblais - Can I get your help to populate the correct life cycle start and end dates for these desktop releases (based on our desktop changelog content) please?

File: source/lifecycle/mattermost-server-releases.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+## Frequency
+Mattermost releases a new server version on the 16th of each month in [binary form](/upgrade/upgrading-mattermost-server). 
+- See the [Mattermost v10 changelog](/lifecycle/mattermost-v10-changelog) for details on what's coming and changing in the next major release. 
+- See the [Mattermost v9 changelog](/lifecycle/mattermost-v9-changelog) for details on v9.x server releases.
+- See the [unsupported Mattermost legacy releases](/lifecycle/unsupported-legacy-releases) documentation for details on older, unsupported  Mattermost releases.
+
+## Latest releases
+
+| **Release** | **Life cycle start** | **Life cycle end** |
```
Comment (cwarnermm) [inline source/lifecycle/mattermost-server-releases.md:19]: @amyblais - Can I get your help to review/correct the life cycle start and end dates for these server releases (based on our v9.x server changelog content) please?

File: source/lifecycle/mattermost-v10-changelog.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+# Mattermost v10 changelog
```
Comment (cwarnermm) [inline source/lifecycle/mattermost-v10-changelog.md:1]: @amyblais - You're welcome to start populating upcoming v10 release details, such as breaking changes or other advanced notifications, on this new v10 changelog page.

File: source/lifecycle/version-archive.rst [hunk 9]
```diff
@@ -0,0 +1,742 @@
+Server Version Archive
```
Comment (cwarnermm) [inline source/lifecycle/version-archive.rst:1]: @nab-77 & @amyblais - Do version archives for desktop app and mobile apps exist? Do we want to start publishing them now/in a future iteration?

File: source/lifecycle/mattermost-mobile-releases.md (hunk 5 above)
Comment (amyblais) [reply source/lifecycle/mattermost-mobile-releases.md:18]: Done for v2.x version - propose we remove v1.x versions as they are no longer supported.

Comment (cwarnermm) [reply source/lifecycle/mattermost-mobile-releases.md:18]: Great idea, @amyblais! Would you be open to removing the v1.x version content?

File: source/lifecycle/mattermost-desktop-releases.md (hunk 6 above)
Comment (amyblais) [reply source/lifecycle/mattermost-desktop-releases.md:22]: Done for v5.x versions. We can probably remove older versions.

File: source/lifecycle/version-archive.rst (hunk 9 above)
Comment (amyblais) [reply source/lifecycle/version-archive.rst:1]: Not currently and I'm not sure if we need it. I'll defer to Neil.

File: source/lifecycle/mattermost-desktop-releases.md (hunk 6 above)
Comment (cwarnermm) [reply source/lifecycle/mattermost-desktop-releases.md:22]: Thank you, @amyblais! Open to pulling the older versions out?

**Review by amyblais — APPROVED** [review]

File: source/lifecycle/client-compatibility.md (hunk 3 above)
Comment (amyblais) [reply source/lifecycle/client-compatibility.md:1]: For reference: Desktop app v5.8.0 (May 16th release) was tested to be compatible with v9.5.0 and newer releases (and v8.1.0 ESR which is now out support) on Linux/Mac/Windows.

**Comment by amyblais** [comment]
@cwarnermm I pushed changes to:
 - Attempted to fix broken links
 - Added compatible server versions for desktop and mobile apps

File: source/lifecycle/version-archive.rst [hunk 10]
```diff
@@ -0,0 +1,742 @@
...
+======================
+
+.. include:: ../_static/badges/allplans-selfhosted.rst
+  :start-after: :nosearch:
+
+.. important::
+  - **Support for Mattermost Server v8.1 `Extended Support Release <https://docs.mattermost.com/deploy/lifecycle/release-policy.html#extended-support-releases>`__ has come to the end of its life cycle on May 15, 2024. Upgrading to Mattermost Server v9.5 Extended Support Release or later is required.**
```
Comment (amyblais) [inline source/lifecycle/version-archive.rst:8]: The ESR link here might still be broken.

**Pull request merged**